src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -2677,11 +2677,6 @@ baddate:
         End Function
 
         Private Shared Function CheckFeatureAvailability(parseOptions As VisualBasicParseOptions, feature As Feature) As Boolean
-            Dim featureFlag = feature.GetFeatureFlag()
-            If featureFlag IsNot Nothing Then
-                Return parseOptions.Features.ContainsKey(featureFlag)
```

Comment:
Same question on `Features` here.

Reply:
Tagging @gafter who mentioned we should not remove too much features machinery, even if no features at the moment.

Reply:
This was actually removed a while back in dev15.5.x as it was unused code once we removed IOperation feature flag. I can revert the changes that removed the machinery with this PR.

Reply:
Note this PR is just cherry picking existing commits that have gone in dev15.5 branch: https://github.com/dotnet/roslyn/pull/22684/commits/27dc60f5f82fff906914b7770b57ca4deaacce5c and https://github.com/dotnet/roslyn/pull/22684/commits/75687b52623f5873991ea5a161cbafb22c81fa82

Reply:
Please do not remove this from code from the compiler. It is intended to facilitate experimental code residing in branches of the compiler, even though the product code does not contain experiments.